server/vulnerabilities/msrc/parsed/full_product_name.go

```diff
@@ -0,0 +1,76 @@
+package msrc_parsed
+
+import "strings"
+
+// FullProdName abstracts a MS full product name.
+// A full product name includes the name of the product plus its arch
+// (if any) and its version (if any).
+type FullProdName string
```

Comment:
this is using a constructor function and seems like a "bag of things", any thoughts on calling this a `Product` instead? which has a `Name()` and `Arch()`.

```go
product := parsed.NewProduct(fullName string)
product.Name()
```

Reply:
Yeah sure, if it makes the code easier to read I'm ok with it.